apps/omg_watcher_rpc/test/omg_watcher_rpc/web/controllers/transaction_test.exs

```diff
@@ -1359,6 +1268,348 @@ defmodule OMG.WatcherRPC.Web.Controller.TransactionTest do
+
+      # Assert when payment amount exactly matched with input amounts
+      params = %{
+        "owner" => Encoding.to_hex(alice.addr),
+        "payments" => [
+          %{"amount" => 30, "currency" => @other_token_hex, "owner" => Encoding.to_hex(bob.addr)}
+        ],
+        "fee" => %{"currency" => @eth_hex}
+      }
+
+      assert %{
+               "transactions" => [
+                 %{
+                   "fee" => %{
+                     "amount" => 5,
+                     "currency" => @eth_hex
+                   },
+                   "inputs" => [
+                     %{"amount" => 20, "currency" => @eth_hex},
+                     %{"amount" => 10, "currency" => @eth_hex},
+                     %{"amount" => 5, "currency" => @eth_hex},
+                     %{"amount" => 30, "currency" => @other_token_hex}
+                   ]
+                 }
+               ]
+             } = WatcherHelper.success?("transaction.create", params)
+
+      # Assert when payment amount exactly matched with input amounts
+      params = %{
+        "owner" => Encoding.to_hex(alice.addr),
+        "payments" => [
+          %{"amount" => 28, "currency" => @other_token_hex, "owner" => Encoding.to_hex(bob.addr)}
+        ],
+        "fee" => %{"currency" => @eth_hex}
+      }
+
+      assert %{
+               "transactions" => [
+                 %{
+                   "fee" => %{
+                     "amount" => 5,
+                     "currency" => @eth_hex
+                   },
+                   "inputs" => [
+                     %{"amount" => 20, "currency" => @eth_hex},
+                     %{"amount" => 10, "currency" => @eth_hex},
+                     %{"amount" => 5, "currency" => @eth_hex},
+                     %{"amount" => 30, "currency" => @other_token_hex}
+                   ]
+                 }
+               ]
+             } = WatcherHelper.success?("transaction.create", params)
+    end
+
+    @tag fixtures: [:phoenix_ecto_sandbox]
+    test "stealth add 1 more input when there're currently 2 inputs and have only 1 utxo left", %{test_server: context} do
+      alice = OMG.TestHelper.generate_entity()
+      bob = OMG.TestHelper.generate_entity()
+      prepare_test_server(context, @fee_response)
+
+      _fee = insert(:txoutput, amount: 5, currency: @eth, owner: alice.addr)
+      _payment = insert(:txoutput, amount: 20, currency: @eth, owner: alice.addr)
+      _merge_1 = insert(:txoutput, amount: 30, currency: @eth, owner: alice.addr)
+
+      params = %{
+        "owner" => Encoding.to_hex(alice.addr),
+        "payments" => [
+          %{"amount" => 20, "currency" => @eth_hex, "owner" => Encoding.to_hex(bob.addr)}
+        ],
+        "fee" => %{"currency" => @eth_hex}
+      }
+
+      assert %{
+               "transactions" => [
+                 %{
+                   "fee" => %{
+                     "amount" => 5,
```

Comment:
Question for all the tests. Anywhere you can access the fee amount from this server in a variable? So that we don't hard-code it?